frontend/packages/topology/src/components/quick-search/QuickSearchController.tsx

```diff
@@ -1,23 +1,63 @@
 import * as React from 'react';
-import { CatalogService } from '@console/dev-console/src/components/catalog/service/CatalogServiceProvider';
+import { useTranslation } from 'react-i18next';
 import { getQueryArgument } from '@console/internal/components/utils';
+import { quickSearch } from './utils/quick-search-utils';
 import QuickSearchButton from './QuickSearchButton';
 import QuickSearchModal from './QuickSearchModal';
+import { QuickSearchData, QuickSearchProviders } from './utils/quick-search-types';
+import { Button } from '@patternfly/react-core';
+import { Link } from 'react-router-dom';
 
-type QuickSearchControllerProps = CatalogService & {
+type QuickSearchControllerProps = {
   namespace: string;
   viewContainer?: HTMLElement;
+  quickSearchProviders: QuickSearchProviders;
+  allItemsLoaded: boolean;
 };
 
 const QuickSearchController: React.FC<QuickSearchControllerProps> = ({
   namespace,
-  searchCatalog,
-  loaded,
+  quickSearchProviders,
   viewContainer,
+  allItemsLoaded,
 }) => {
   const [isQuickSearchActive, setIsQuickSearchActive] = React.useState<boolean>(
     !!getQueryArgument('catalogSearch'),
   );
+  const { t } = useTranslation();
+
+  const searchCatalog = React.useCallback(
+    (searchTerm: string): QuickSearchData => {
+      return quickSearchProviders.reduce(
+        (acc, quickSearchProvider) => {
+          const items = quickSearchProvider.loaded
+            ? quickSearch(quickSearchProvider.items, searchTerm)
+            : [];
+          const itemCount = items.length;
+          const viewAllLink =
+            itemCount > 0
+              ? [
+                  <Button variant="link" isInline>
```

Comment:
This would prevent tab to go to the button and will directly jump to link.
```suggestion
                  <Button variant="link" tabIndex={-1} isInline>
```